cabal-install/src/Distribution/Client/Dependency.hs

```diff
@@ -819,11 +830,17 @@ resolveDependencies platform comp pkgConfigDB params =
                     solveExes
                     order
                     verbosity
+                    outputJson
                   ) =
         if asBool (depResolverAllowBootLibInstalls params)
           then params
           else dontInstallNonReinstallablePackages params
 
+    formatProgress :: Progress SolverTrace String a -> Progress String String a
+    formatProgress p = foldProgress (\x xs -> Step (formatter x) xs) Fail Done p
+      where
+        formatter = if outputJson then error "FIXME: To be implemented ..." else show
```

Comment:
I don't think this should rely on the `Show` instance, but rather than explicit function `renderSolverTrace :: SolverTrace -> String`.. where the intent is clearly to display the message to the user.

Reply:
I think I came up with the idea of using Show, but, thinking again, I agree with your comment.